### PR TITLE
PERL-799: parse Logical session timeout from isMaster

### PR DESCRIPTION
**Pull request by TBSliver**
Includes a skipped test spec that I am unsure if applies to this driver, with the PossiblePrimary option.

### CODE REVIEW

**Comment by xdg** [comment]
Updated my comment with a better example.

**Comment by TBSliver** [comment]
Right have updated the logic - I think I may have misunderstood part of the spec originally.

Note that instead of using a -1 for a signalling value, I've used undef - as this is the value given in the spec, and used in the spec json files, so saves reworking all of those.

I've also added an `is_data_bearing` boolean to the server as suggested :D

File: lib/MongoDB/_Topology.pm [hunk 1]
```diff
@@ -562,23 +562,17 @@ sub _check_wire_versions {
...
-        }
-        $self->_set_logical_session_timeout_minutes( $timeout );
+    my @data_bearing_servers = grep { $_->is_data_bearing } $self->all_servers;
+    my $timeout = min map {
+        # use -1 as a flag to prevent undefined warnings
+        defined $_->logical_session_timeout_minutes
+          ? $_->logical_session_timeout_minutes
+          : -1
+    } @data_bearing_servers;
+    if ( defined $timeout && $timeout < 0 ) {
```
Comment (xdg) [inline lib/MongoDB/_Topology.pm]: I think $timeout would only be undef here if there were no data-bearing servers, right?  If that's right, please add a comment to that effect for clarity.

**Comment by xdg** [comment]
Awesome!  Rebased, squashed and merged!

**Pull request closed without merging**